Script/Selenops.swift

```diff
@@ -9,76 +9,130 @@
 import Foundation
 
 // Input your parameters here
-let startUrl = URL(string: "https://developer.apple.com/swift/")!
-let wordToSearch = "Swift"
-let maximumPagesToVisit = 10
+var startUrl = URL(string: "https://developer.apple.com/swift/")!
+var wordToSearch = "Swift"
+var maximumPagesToVisit = 10
 
 // Crawler Parameters
 let semaphore = DispatchSemaphore(value: 0)
 var visitedPages: Set<URL> = []
-var pagesToVisit: Set<URL> = [startUrl]
+var pagesToVisit: Set<URL> = [startUrl] //originally set this to startURL, set below for dynamic page setting
 
 // Crawler Core
 func crawl() {
-  guard visitedPages.count <= maximumPagesToVisit else {
-    print("🏁 Reached max number of pages to visit")
-    semaphore.signal()
-    return
-  }
-  guard let pageToVisit = pagesToVisit.popFirst() else {
-    print("🏁 No more pages to visit")
-    semaphore.signal()
-    return
-  }
-  if visitedPages.contains(pageToVisit) {
-    crawl()
-  } else {
-    visit(page: pageToVisit)
-  }
+    guard visitedPages.count <= maximumPagesToVisit else {
```

Comment:
Hi Matthew,

pretty much everywhere the indentation has changed from double space to tabs/four spaces, can you please revert this?

It's hard to spot where the code has actually changed otherwise 😅